### ISSUE TITLE
Are there any command for changing branch setting??

### ISSUE THREAD

**Issue by AkhilaDamera** [ngetchell/PSGitLab#187]
Hi,

After the project creation, I need to create the readme.md file, master branch and change the master branch setting to private/protected. Are there any commands available to initiate the repo and change the master branch settings??

Appreciate your response,

Thanks,
akhila

**Comment by it-praktyk** [comment]
You have to create a repository, the README.md file locally, commit changes and push them to GitLab.

You can change the visibility of a project using [Set-GitLabProject](`https://github.com/ngetchell/PSGitLab/blob/master/PSGitLab/Public/Projects/Set-GitLabProject.ps1)` function.

**Comment by AkhilaDamera** [comment]
I can add file locally and commit to GitLab, I need to automate the process of creating repository and adding the readme file. Do you have any functions available for this??


The Set-GitLabProject is for renaming correct? But,I need to change the branch permission under a repository.

**Comment by it-praktyk** [comment]
> I can add file locally and commit to GitLab, I need to automate the process of creating repository and adding the readme file. Do you have any functions available for this??

Nope.

The GitLab [Branches API](https://docs.gitlab.com/ee/api/branches.html) is not implemented in the PSGitLab yet.

> The Set-GitLabProject is for renaming correct? 

Nope, the parameter Visibility is for that.

> But,I need to change the branch permission under a repository.

Do you mean to protect a branch?

**Comment by AkhilaDamera** [comment]
Yes. I need to change the branch setting to protected and I believe that comes under the Branches API

**Comment by AkhilaDamera** [comment]
Any idea when would the PSGitLab support the Branches API feature??

**Comment by robinmalik** [comment]
There is no roadmap. All additions/improvements are done by the community/public when motivated and available to do so.

**Comment by ngetchell** [comment]
@it-praktyk and @robinmalik are correct. No timeline exists for support for the Branches API. My goal in the project was to cover the APIs I need but make it easy for others to add what they need. 

I would welcome any addition from @AkhilaDamera to add support. I'd be more than happy to aid any attempt if you've never done anything like this.
